osu.Game/Screens/Ranking/Expanded/ExpandedPanelMiddleContent.cs

```diff
@@ -65,10 +65,14 @@ private void load(BeatmapDifficultyCache beatmapDifficultyCache)
             var metadata = beatmap.BeatmapSet?.Metadata ?? beatmap.Metadata;
             string creator = metadata.Author.Username;
 
+            bool isPerfect = true;
+            isPerfect &= !score.Statistics.TryGetValue(HitResult.Miss, out int missCount) || missCount == 0;
+            isPerfect &= !score.Statistics.TryGetValue(HitResult.LargeTickMiss, out int largeTickMissCount) || largeTickMissCount == 0;
```

Comment:
Might be a good point to make a helper method on `ScoreInfo` to do this.

Reply:
Yeah was double-minded on this one, will do it then.